screens/EditOffer.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { View, ScrollView, AsyncStorage } from 'react-native'
+import { LocaleContext } from '../locales/LocaleContext'
+import { DismissKeyboard } from '../components/DismissKeyboard'
+import ScreenHeader from "../components/ScreenHeader";
+import { getOrderOffer, getOffers, clearOrderOffer, removeDuplicate } from '../actions'
+import styles from '../styles'
+import OfferForm from './OfferForm'
+import { api, dispatchFetchRequest, successMessage } from '../constants/Backend'
+import LoadingScreen from "./LoadingScreen";
+import moment from "moment"
+
+class EditOffer extends Component {
+  static navigationOptions = {
+    header: null
+  }
+  static contextType = LocaleContext
+
+  state = {
+    startDate: '',
+    endDate: '',
+    products: [],
+    productIds: [],
+    productLabelIds: [],
+    uniqueProductLabelIds: []
+  }
+
+  componentDidMount() {
+    this.props.getOrderOffer(this.props.navigation.state.params.offerId);
+  }
+
+  handleonChange = (data) => {
+    let startDate = moment(data.date).format('YYYY-MM-DD') + 'T' + moment(data.time).format('HH:mm:ss');
+    let endDate = moment(data.todate).format('YYYY-MM-DD') + 'T' + moment(data.totime).format('HH:mm:ss')
+    this.setState({
+      startDate: startDate,
+      endDate: endDate,
+      products: data.products
+    });
+  }
+
+  handleSubmit = (values) => {
+    let productLabelIds = [];
+    let uniqueProductLabelIds = [];
+    values.productIds = [];
+    values.productLabelIds = [];
+    var offerId = this.props.navigation.state.params.offerId;
+    const { products } = this.state
+
+    values.offerType !== 0 && values.appliesToAllProducts === undefined ? values.appliesToAllProducts = false : null;
+    if (Platform.OS !== 'ios') {
+      values.startDate = this.state.startDate;
+      values.endDate = this.state.endDate;
+    }
+
+    products !== undefined && products !== false && products.map(selectedProduct => {
+      values.productIds.push(selectedProduct.productId)
+      productLabelIds.push(selectedProduct.labelId)
+      uniqueProductLabelIds = productLabelIds.filter(function (item, index) {
+        return productLabelIds.indexOf(item) == index;
+      })
+      values.productLabelIds = uniqueProductLabelIds;
+    })
+
+    if (values.offerType === "PRODUCT") {
+      values.offerType = 1
```

Comment:
why is offerType's value = 1 or 2? It should be either PRODUCT or ORDER.

Reply:
this is not for saving, showing edit result in segment control.